src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -1442,6 +1473,23 @@ void ImageReplayer<I>::handle_shut_down(int r) {
 }
 
 template <typename I>
+void ImageReplayer<I>::handle_remote_journal_metadata_updated() {
+  dout(20) << dendl;
+
+  cls::journal::Client client;
+  int r = m_remote_journaler->get_cached_client(m_local_mirror_uuid, &client);
+  if (r < 0) {
+    derr << "failed to retrieve client: " << cpp_strerror(r) << dendl;
+    return;
+  }
+
+  if (client.state != cls::journal::CLIENT_STATE_CONNECTED) {
+    dout(0) << "client flagged disconnected, stopping image replay" << dendl;
+    stop(nullptr, false, "disconnected");
```

Comment:
Same comment here: should we set an error code so that an admin can see the problem?